resource/js/util/PreProcessor/Linker.js

```diff
@@ -1,10 +1,23 @@
+var path = require('path');
 
 export default class Linker {
   process(markdown) {
 
     return markdown
       //.replace(/\s(https?:\/\/[\S]+)/g, ' <a href="$1">$1</a>') // リンク
       .replace(/\s<((\/[^>]+?){2,})>/g, ' <a href="$1">$1</a>') // ページ間リンク: <> でかこまれてて / から始まり、 / が2個以上
+      // Pukiwiki Like Linker
+      // see: https://regex101.com/r/k2dwz3/3
+      .replace(/\[\[(([^(\]\])]+)>)?(.+?)\]\]/g, function(all, group1, group2, group3) {
+        // create url
+        // http(s) から始まっていればそのまま利用、そうでなければ window.location.pathname と連結
+        var url = (group3.match(/^https?:\/\//)) ? group3 : path.join(window.location.pathname, group3);
```

Comment:
> if the page is/foo/bar/baz and has ./hoge
> 
>    * it should link to /foo/bar/hoge
>    * it works on a single page

First, the motivation of this PR is that we want to use flexible Pukiwiki like path resolution.

https://pukiwiki.osdn.jp/?PukiWiki%E3%81%AE%E4%BD%BF%E3%81%84%E6%96%B9#jee7c1ee

So if the page is /foo/bar/baz,
`[[./hoge]]` generates the link to `/foo/bar/baz/hoge` (not `/foo/bar/hoge`).
This is the intended behavior.

> if user access to /foo/ (a list page)
> 
>   * and show timeline view,
>   * then, the link may point to /foo/hoge

For the above behavior doesn't cause such a problem.